blockchain/ethereum.go

```diff
@@ -347,7 +347,7 @@ func (e *EthereumClient) ProcessTransaction(tx *types.Transaction) error {
 func (e *EthereumClient) ProcessEvent(name string, event *types.Log, confirmedChan chan bool, errorChan chan error) error {
 	var eventConfirmer HeaderEventSubscription
 	if e.GetNetworkConfig().MinimumConfirmations == 0 {
-		eventConfirmer = &InstantConfirmations{}
+		eventConfirmer = NewL2TxConfirmer(e, event.TxHash)
 	} else {
 		eventConfirmer = NewEventConfirmer(name, e, event, e.GetNetworkConfig().MinimumConfirmations, errorChan, confirmedChan)
```

Comment:
What is the logic behind having both TxConfirmer and EventConfirmer determined because of `MinimumConfirmations`? What that parameter then stands for? Tx confirmations or events?

Reply:
We can tweak that, but I think the idea is that for anything on-chain to count as "real", it has to pass this minimum confirmation count.